src/Resources/contao/classes/BackendUser.php

```diff
@@ -111,6 +123,32 @@ protected function __construct()
 		$this->strHash = \Input::cookie($this->strCookie);
 	}
 
+	public static function getInstance()
+	{
+		/** @var TokenInterface $token */
+		$token = \System::getContainer()->get('security.token_storage')->getToken();
+
+		// Try to load token from security storage
+		if ($token !== null && is_a($token->getUser(), static::class))
+		{
+			return $token->getUser();
+		}
+
+		/** @var SessionInterface $session */
+		$session = \System::getContainer()->get('session');
+
+		// Try to load possibly authenticated BackendUser from session
+		if ($session->has(self::SECURITY_SESSION_KEY) && $token = unserialize($session->get(self::SECURITY_SESSION_KEY)))
```

Comment:
Maybe we should add `array('allowed_classes' => TokenInterface::class)` as second argument for `unserialize()`?

Reply:
👍

Reply:
Be aware that there can be other classes. The user in the token is serialized as well. And you never know what else the token contains (e.g. a `FacebookToken` could have other dependencies).

Reply:
But that as well should implement the `TokenInterface`, or do I miss something here?

Reply:
You can't provide interfaces nor parent classes for `allowed_classes`.

Reply:
> But that as well has to implement the TokenInterface, or do I miss something here?

A user can be a `UserInterface` or what do you mean?